### ISSUE TITLE
Lint RHS literals in conditional expressions?

### ISSUE THREAD

**Issue by IndrajeetPatil** [r-lib/lintr#1835]
## Preamble

It can be difficult to understand conditional expressions when the numeric, string, etc. literals appear in RHS of the expression, and `{lintr}` can help detect such instances.

It's rare to see code like this, so I am wondering if it is worth introducing a new linter, but wanted to create an issue nonetheless.

## Examples

Actual

```r
if (0L == length(x)) 1L
if ("a" == x) 1L
...
```

Suggested/Expected

```r
if (length(x) == 0L) 1L
if (x == "a") 1L
...
```

Exceptions

If there are literals on both RHS and LHS. E.g.

```r
dplyr::filter(data, 0L == 1L)
```

**Comment by Bisaloo** [comment]
This sounds related to the `yoda_test_linter()`. Would it be worth adding this new functionality there (and rename it with a more generic name maybe?)?

**Comment by IndrajeetPatil** [comment]
You are right! This would be a perfect for that linter. And, yes, we will need to change the name of this linter. 

How about `yoda_condition_linter()` (https://en.wikipedia.org/wiki/Yoda_conditions)? Generic enough to cover tests, expressions, etc.

**Comment by MichaelChirico** [comment]
yes I had this in mind as going into yoda_test_linter()

**Comment by MichaelChirico** [comment]
This covers a TODO as well:

https://github.com/r-lib/lintr/blob/72eeefd87596ea7fa72ee1a8b14bcb82d322c381/tests/testthat/test-yoda_test_linter.R#L73-L75
